src/transformers/models/llava/configuration_llava.py

```diff
@@ -147,6 +147,10 @@ def vocab_size(self):
         )
         return self._vocab_size
 
+    @vocab_size.setter
+    def vocab_size(self, value):
+        self._vocab_size = value
+
```

Comment:
Is this good @NielsRogge @amyeroberts ?

Reply:
Shouldn't we remove the `@property` annotator above? That fixed https://github.com/huggingface/transformers/issues/29789 for me. Isn't property for immutable things (whereas the vocabulary size can change? cc @amyeroberts

Reply:
Properties aren't for immutable things (this is the value of a setter). The property enables us to emit a deprecation warning if the property is accessed as an attribute.

Removing the property annotation would mean that previous usage of `config.vocab_size` would break, and people would have to use `config.vocab_size()`